scripts/cmake/vcpkg_replace_string.cmake

```
@@ -1,5 +1,16 @@
 function(vcpkg_replace_string filename match replace)
+    cmake_parse_arguments(PARSE_ARGV 3 "arg" "REGEX;IGNORE_UNCHANGED" "" "")
+    if (arg_REGEX)
+        set(arg_REGEX "REGEX")
+    else ()
+        set(arg_REGEX "")
+    endif ()
     file(READ "${filename}" contents)
-    string(REPLACE "${match}" "${replace}" contents "${contents}")
+    string(SHA512 before_hash "${contents}")
+    string(${arg_REGEX} REPLACE "${match}" "${replace}" contents "${contents}")
+    string(SHA512 after_hash "${contents}")
+    if (NOT arg_IGNORE_UNCHANGED AND before_hash STREQUAL after_hash)
+        message(WARNING "vcpkg_replace_string made no changes.")
```

Should this be:

```suggestion
        message("${Z_VCPKG_BACKCOMPAT_MESSAGE_LEVEL}" "vcpkg_replace_string made no changes.")
```

so that `--enforce-port-checks` elevates it to an error?

---

(Note that this will imply that existing cases in the registry need to be fixed)

---

Awesome, I changed it as suggested.

---

> (Note that this will imply that existing cases in the registry need to be fixed)

I don't see a problem with this, I think adding an explicit `IGNORE_UNCHANGED` as the fix is still better if it can be justified. It draws attention to the potential issues when modifying or at least highlights areas for improvement.

---

This will require adding the `IGNORE_UNCHANGED` to a bunch of ports and probably some helper scripts, I'll convert this PR to draft while those get fixed.

---

@vicroms how do you see this process unfolding?
If it is something that's overlooked, I'd imagine that at least the ports with `"maintainers": ""` might want to change as appropriate rather than just ignoring it.
Otherwise, I guess I could just make the changes to let the CI pass and add them to another commit to this PR.
Thoughts?

---

If the CI revealed all the errors there was only one port with a maintainer so I just made changes to all of them.

---

@gerard-ryan-immersaview thanks for taking the time to fix all these calls

---

Looks like it might take a couple of rounds, CI seems to be bringing up different ones this time.

---

Yeah, cascade failures are hidden until their dependencies' failures are resolved.

---

@gerard-ryan-immersaview 

I'm bringing this PR for discussion with the rest of the team. The number of ports that need to be fixed is greater that the team expected, so we need to discuss if making the warning a failure is actually a good decision or if we should think of a different way.